### PR TITLE
Fixes for fluidsynth >= 2.0.0 V2

### PR DESCRIPTION
**Pull request by schnitzeltony**
It needs a new pull request for develop branch. Will close the other one

V1 -> V2:
* rebase against develop branch
* rename function & vars
* add missing 'sbFluidDefaultsStored = true;'

### CODE REVIEW

File: source/backend/plugin/CarlaPluginFluidSynth.cpp [hunk 1]
```diff
@@ -1683,13 +1770,19 @@ class CarlaPluginFluidSynth : public CarlaPlugin
     float** fAudio16Buffers;
     float   fParamBuffers[FluidSynthParametersMax];
 
+    static bool  sbFluidDefaultsStored;
```
Comment (falkTX) [inline source/backend/plugin/CarlaPluginFluidSynth.cpp]: I meant before to use "s" prefix *instead* of "b" or "f", just a single one.
Rest looks quite fine, thanks!

File: source/backend/plugin/CarlaPluginFluidSynth.cpp [hunk 2]
```diff
@@ -1656,6 +1689,60 @@ class CarlaPluginFluidSynth : public CarlaPlugin
     }
 
 private:
+    void initializeFluidDefaultsIfNeeded()
+    {
+        if (!sbFluidDefaultsStored)
```
Comment (falkTX) [inline source/backend/plugin/CarlaPluginFluidSynth.cpp]: please revert this check, like so:
```
if (sFluidDefaultsStored)
    return;
sFluidDefaultsStored = true;
// ...
```
The purpose is to continue the main code without extra indentation

**Comment by schnitzeltony** [comment]
V3: 
* rebased to develop branch
* variable rename
* if bails out in case default were loaded already

**Comment by falkTX** [comment]
Very nice, thanks!

**Pull request merged**